Sources/Apollo/HTTPNetworkTransport.swift

```diff
@@ -69,18 +69,23 @@ public class HTTPNetworkTransport: NetworkTransport {
   ///
   /// - Parameters:
   ///   - operation: The operation to send.
+  ///   - fetchOptions: The HTTP Method to be used.
   ///   - completionHandler: A closure to call when a request completes.
   ///   - response: The response received from the server, or `nil` if an error occurred.
   ///   - error: An error that indicates why a request failed, or `nil` if the request was succesful.
   /// - Returns: An object that can be used to cancel an in progress request.
-  public func send<Operation>(operation: Operation, completionHandler: @escaping (_ response: GraphQLResponse<Operation>?, _ error: Error?) -> Void) -> Cancellable {
+  public func send<Operation>(operation: Operation, fetchOptions: FetchOptions, completionHandler: @escaping (_ response: GraphQLResponse<Operation>?, _ error: Error?) -> Void) -> Cancellable {
+    let body = requestBody(for: operation)
     var request = URLRequest(url: url)
-    request.httpMethod = "POST"
     
+    if fetchOptions.rawValue == "GET", let urlForGet = mountUrlWithQueryParamsIfNeeded(body: body) {
+        request = URLRequest(url: urlForGet)
+    } else if fetchOptions.rawValue == "POST" {
+        request.httpBody = try! serializationFormat.serialize(value: body)
```

Comment:
I would reeeeeally avoid `try!` here, as it'll crash an app if it fails. 

What about a `do/try` and call the completion closure with any caught error?

Reply:
Agreed. Actually, I did not change this line, just put it inside my `if` (now a switch as you suggested). But I agree that it could be changed.

Reply:
Looks like it actually was changed when I look at the diff - there's still a `try` but that's wayyy better than a `try!`, since it catches and returns the error instead of crashing the app.